src/logging/transformers/stripAnsi.ts

```diff
@@ -0,0 +1,63 @@
+import { TransformableInfo } from 'logform';
+import { format } from 'winston';
+
+/**
+ * Regex pattern to match ANSI characters.
+ */
+const pattern = [
+	'[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?\\u0007)',
+	'(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-ntqry=><~]))',
+].join('|');
+
+/**
+ * RegExp for global matching ANSI characters.
+ */
+const ansiRegex = new RegExp(pattern, 'g');
+
+/**
+ * Remove ANSI characters from a string by replacing them with empty an empty string.
+ *
+ * @param text
+ */
+function ansiReplace(text: unknown): typeof text {
+	return typeof text === 'string' ? text.replace(ansiRegex, '') : text;
+}
+
+/**
+ * Strip ANSI characters from strings in simple, arbitrary data. N.B. this is not
+ * hardened to work with any possible javascript type and is only meant for arrays,
+ * basic objects and strings.
+ *
+ * @param data arbitrary data
+ */
+function sanitizeAnsi(data: unknown): unknown {
```

Comment:
I think `sanitize` is the wrong verb here: it's usually used to guard against security hazards. We're really stripping out data we don't want here, but not for security reasons. Can it be named `stripAnsiShellCodes` or some such?

It is called recursively (unbounded). I don't think that is a big concern, but maybe worth pondering for a second if users have any way to submit a malicious payload that would crash the server?